### ISSUE TITLE
[RPG project] Potential accessibility issue in step 138 and 142

### ISSUE THREAD

**Issue by huyenltnguyen** [freeCodeCamp/freeCodeCamp#52775]
### Describe the Issue

In step 138 and 142, campers are asked to create an object, and one of its values must contain an emoji (☠️ or 🎉).

I tested the instruction with VoiceOver, and it was announced as follows:

| Step | Original instruction | VoiceOver interpretation |
| --- | --- | --- |
| 138 | "and set `text` to `You die. ☠️`" | "and set `text` to `You die skeleton crossbones.`" |
| 142 | "and the `text` should be `You defeat the dragon! YOU WIN THE GAME! 🎉`" | "and the `text` should be `You defeat the dragon! You win the game! party popper`" |

The challenges won't pass if the string doesn't contain the emoji or if it doesn't follow the exact casing.

My concerns are:
- I'm not sure if non-sighted users can tell they need to insert an emoji here. I guess they might, if they're familiar enough with the emoji names
- Screen readers apparently don't differentiate between lowercase and uppercase when reading text, so it could be confusing for non-sighted users if they input the correct message but in lowercase and the tests don't pass. But maybe I'm overthinking here, this is already the case for any scenarios that require case-sensitive input (camel case in variable name, variable name vs class name, etc.)

### Affected Page

- https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-basic-javascript-by-building-a-role-playing-game/step-138
- https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-basic-javascript-by-building-a-role-playing-game/step-142

### Expected behavior

Re: emojis, probably one of these:
- Add a visually-hidden note to the instruction to inform non-sighted users that they are going to use emojis
- Remove the emojis from the instructions and the tests
- Update the tests to ignore the absence of emojis

As for the case sensitivity, we can make the tests case-insensitive if we think it can trip up the users.

**Comment by bbsmooth** [comment]
Regarding case and screen readers, I don't think this is something we need to be concerned about. Screen readers offer various levels of support for differentiating upper and lower case letters. For example, some of them will announce an uppercase letter with a higher pitch. In NVDA, you can configure the screen reader to either say the word "cap" before a capital letter or add a beep before a capital letter. And of course, you can just copy/paste the string directly from the instructions which will preserve the case. And as you mentioned above, case sensitivity is a thing in programming, and thus it can't be avoided, so we probably won't be doing screen reader users any favors by erasing this distinction.

Regarding emojis, I understand what you are saying here. The emoji is read as if it were text, so it might not be clear that an actual emoji is required to pass the test. I think one way we can make this more accessible is to make it clear in the instructions that an emoji is being used at the end of the text. For example, for step 138:

> In the `locations` array, add another object at the end. Set the `name` property to `lose`, set `button text` to an array with three `REPLAY?` strings, set `button functions` to an array with three `restart` variables, and set `text` to `You die. ☠️`. 

> Note: To insert the skull and crossbones emoji at the end of the text property, you can either copy/paste it from these instructions or use the HTML code `&#x2620;`.

This also has the additional advantage of giving people an option other than copy/paste to pass the step, which is a big accessibility win if you ask me.

**Comment by a2937** [comment]
Actually there is a slight issue. Something is preventing the challenges from rendering the HTML text "&#x1F389;" as the party popper. Or the skull and crossbones from appearing on the page.

![image](https://github.com/freeCodeCamp/freeCodeCamp/assets/9373245/d8223d7a-3972-40fd-9c4d-a7d87dc37236)

**Comment by a2937** [comment]
I  do have something resembling a theory though nothing concrete unfortunately. My first thought it had to do with the way emojis are being handled by the website.

**Comment by gikf** [comment]
That's due to changing `text.innerText` in `update` function, once that's changed to `text.innerHTML` the icon is displayed.

**Comment by a2937** [comment]
I've been searching all over the code base and haven't found anything resembling what you said yet. Can I have some help looking for it? I've checked `frame.ts` and `editor.ts` but nothing there.

**Comment by gikf** [comment]
It's not in the codebase per se, but in this project
https://github.com/freeCodeCamp/freeCodeCamp/blob/b95cc96913293808ce52caf2bf101a1ec99558d8/curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a94114ce0b8918b487390f.md?plain=1#L223-L232

**Comment by a2937** [comment]
Oh. I see. Hmmm...how do you suggest I handle this? After all `text.innerText` was introduced all the way back in step 46 and the where the icons start is step 138.

**Comment by gikf** [comment]
If it would be desired I think this could be taken care of as a demonstration of difference between `innerText` and `innerHTML`. Starting with the former that's not displayed as intended and changing it to the latter.

Even better would be if it would happen in a way that doesn't require camper to go through game to see how it is displayed.

**Comment by a2937** [comment]
I agree. Adding a new brand new step to the curriculum after step 138 dealing with changing `text.innerText` to `text.innerHtml` to resolve this issue would be really nice.

**Comment by a2937** [comment]
What should we do about my PR for now?

**Comment by naomi-lgbt** [comment]
> I agree. Adding a new brand new step to the curriculum after step 138 dealing with changing text.innerText to text.innerHtml to resolve this issue would be really nice.

This is a good idea, let's move forward with this.
